tests/autopilot_pool_expand/aut_pool_expand_test.go

```diff
@@ -0,0 +1,247 @@
+				},
+			},
+		},
+	},
+}
+
+var ruleResizeBy50IfPoolTotalSizeLessThan300ResizeDisk = apapi.AutopilotRule{
+	ObjectMeta: meta_v1.ObjectMeta{
+		Name: "storage-pool-resize-disk-300",
+	},
+	Spec: apapi.AutopilotRuleSpec{
+		Selector: apapi.RuleObjectSelector{
+			LabelSelector: meta_v1.LabelSelector{
+				MatchLabels: map[string]string{
+					"name": "storage-pool-resize",
+				},
+			},
+		},
+		Conditions: apapi.RuleConditions{
+			Expressions: []*apapi.LabelSelectorRequirement{
+				{
+					Key:      portworx.PxPoolTotalCapacityMetric,
+					Operator: apapi.LabelSelectorOpLt,
+					Values:   []string{"300"},
+				},
+			},
+		},
+		Actions: []*apapi.RuleAction{
+			{
+				Name: portworx.StorageSpecAction,
+				Params: map[string]string{
+					portworx.RuleActionsScalePercentage: "50",
+					portworx.RuleScaleType:              portworx.RuleScaleTypeResizeDisk,
+				},
+			},
+		},
+	},
+}
+
+var autopilotruleBasicTestCases = []apapi.AutopilotRule{
+	ruleResizeBy50IfPoolAvailableUsageMoreThan70AddDisk,
+	ruleResizeBy50IfPoolAvailableUsageMoreThan70ResizeDisk,
+}
+
+func TestAutoPilot(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	var specReporters []Reporter
+	junitReporter := reporters.NewJUnitReporter("/testresults/junit_autopilot.xml")
+	specReporters = append(specReporters, junitReporter)
+	RunSpecsWithDefaultAndCustomReporters(t, "Torpedo : Autopilot", specReporters)
+}
+
+var _ = BeforeSuite(func() {
+	InitInstance()
+})
+
+// This testsuite is used for performing basic scenarios with Autopilot rules where it
+// schedules apps and wait until workload is completed on the volumes and then validates
+// storage pool(s) size(s) on the nodes where volumes reside
+var _ = Describe(fmt.Sprintf("{%sBasicAddDisk}", testNameSuite), func() {
+	It("has to fill up the volume completely, resize the storage pool(s), validate and teardown apps", func() {
+		var err error
+		testName := strings.ToLower(fmt.Sprintf("%sBasicAddDisk", testNameSuite))
+		appWorkloadSize := uint64(16106127360) //15 Gb
+		apRules := autopilotruleBasicTestCases
+		var contexts []*scheduler.Context
+
+		workerNodes := node.GetWorkerNodes()
+		for _, workerNode := range workerNodes {
+			err = AddLabelsOnNode(workerNode, apRules[0].Spec.Selector.LabelSelector.MatchLabels)
+			Expect(err).NotTo(HaveOccurred())
+		}
+
+		Step("schedule applications", func() {
+			for i := 0; i < Inst().ScaleFactor; i++ {
+				taskName := fmt.Sprintf("%s-%v", fmt.Sprintf("%s-%d", testName, i), Inst().InstanceID)
+				context, err := Inst().S.Schedule(taskName, scheduler.ScheduleOptions{
+					AppKeys:            Inst().AppList,
+					StorageProvisioner: Inst().Provisioner,
+					ScaleStorageFactor: Inst().ScaleStorageFactor,
+				})
+				Expect(err).NotTo(HaveOccurred())
+				Expect(context).NotTo(BeEmpty())
+				contexts = append(contexts, context...)
+			}
+		})
+
+		appEnvs := []scheduler.EnvVar{
+			scheduler.EnvVar{
+				Name:  "SIZE",
```

Comment:
"SIZE" needs to be a `const` as it is a special variable and used at multiple places